library/spdm_responder_lib/libspdm_rsp_subscribe_event_types_ack.c

```diff
@@ -0,0 +1,150 @@
+                                               SPDM_ERROR_CODE_UNSUPPORTED_REQUEST,
+                                               SPDM_SUBSCRIBE_EVENT_TYPES,
+                                               response_size, response);
+    }
+    if (!libspdm_is_capabilities_flag_supported(
+            spdm_context, false,
+            0, SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_EVENT_CAP)) {
+        return libspdm_generate_error_response(
+            spdm_context, SPDM_ERROR_CODE_UNSUPPORTED_REQUEST,
+            SPDM_SUBSCRIBE_EVENT_TYPES, response_size, response);
+    }
+    if (!spdm_context->last_spdm_request_session_id_valid) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_SESSION_REQUIRED, 0,
+                                               response_size, response);
+    }
+    session_id = spdm_context->last_spdm_request_session_id;
+    session_info = libspdm_get_session_info_via_session_id(spdm_context, session_id);
+    if (session_info == NULL) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_SESSION_REQUIRED, 0,
+                                               response_size, response);
+    }
+    session_state = libspdm_secured_message_get_session_state(
+        session_info->secured_message_context);
+    if (session_state != LIBSPDM_SESSION_STATE_ESTABLISHED) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_UNEXPECTED_REQUEST, 0,
+                                               response_size, response);
+    }
+
+    /* -=[Validate Request Phase]=- */
+    if (request_size < sizeof(spdm_message_header_t)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                               response_size, response);
+    }
+    if (spdm_request->header.spdm_version != libspdm_get_connection_version(spdm_context)) {
+        return libspdm_generate_error_response(spdm_context,
+                                               SPDM_ERROR_CODE_VERSION_MISMATCH, 0,
+                                               response_size, response);
+    }
+
+    /* This message can only be in secured session.
+     * Thus don't need to consider transport layer padding, just check its exact size. */
+    subscribe_event_group_count = spdm_request->header.param1;
+
+    if (subscribe_event_group_count == 0) {
+        if (request_size != sizeof(spdm_message_header_t)) {
+            return libspdm_generate_error_response(spdm_context,
+                                                   SPDM_ERROR_CODE_INVALID_REQUEST, 0,
+                                                   response_size, response);
+        }
+    } else {
+        if (request_size < (sizeof(spdm_message_header_t) + sizeof(uint32_t))) {
```

Comment:
`sizeof(spdm_message_header_t) + sizeof(uint32_t)` can be `sizeof(spdm_subscribe_event_types_request_t)`